File: basics/Sorting/Python/BubbleSort.py
"""
Bubble sort algorithm

Basic example of Bubble Sort
Sample array: 23, 8, 15, 17, 4, 40, 11, 31
Complexity: O (2^n)
"""

def bubleSort(dataset):
	#TODO
	for i in range(len(dataset)-1,0,-1):
		for j in range(i):
			if dataset[j] > dataset[j+1]:
				temp = dataset[j]
				dataset[j] = dataset[j+1]
				dataset[j+1] = temp

	#	print ("Current state: ", dataset)



def main():
	list1 = [6,39,90,13,2,96,87,36,41]
	bubleSort(list1)
	print("Result: ", list1)

if __name__ == "__main__":
	main()

File: basics/Sorting/Javascript/BubbleSort.js
/*
 * Basic example of Bubble Sort
 *
 * Sample array: 23, 8, 15, 17, 4, 40, 11, 31
 * Complexity: O (2^n)
 */
 
function sortBubble(numbers) {

    for(i=numbers.length-1; i>=0;i--) {
        for(j=0; j < numbers.length-1; j++) {
            if(numbers[j] > numbers[j+1]) {
                temp = numbers[j];
                numbers[j] = numbers[j+1];
                numbers[j+1] = temp;
            }
        }
        //console.log("Current state: " + numbers);
    }

    return numbers;
}

var numbers = [6,39,90,13,2,96,87,36,41];
result = sortBubble(numbers)
console.log("Result: " + result)